pkg/util/deployer/deployer.go

```diff
@@ -127,6 +137,47 @@ func (depl *deployer) Remove(ctx context.Context, data interface{}) error {
 	return nil
 }
 
+func (depl *deployer) removeOne(ctx context.Context, obj kruntime.Object) (string, error) {
+	// remove everything we created that has name and ns
+	nameValue, err := getField(obj, "Name")
+	if err != nil {
+		return "", err
+	}
+	nsValue, err := getField(obj, "Namespace")
+	if err != nil {
+		return "", err
+	}
+	name := nameValue.String()
+	ns := nsValue.String()
+	if reflect.TypeOf(obj).String() == "*v1.Namespace" {
```

Comment:
Can we instead compare against `obj.GetObjectKind().GroupVersionKind().GroupKind()` here?